### PR TITLE
fix: some minor bugs + improvements

### PR DESCRIPTION
**Pull request by phanhuynhquy**
### [Change Description]
Fix some minor bugs + improvements related to error handling

### [Ticket]
None

### [Test]
run again tests

### [CheckList]
None

### CODE REVIEW

File: tests/replicate_automatically_test.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func TestReplicateAutomatically(t *testing.T) {
 		defer db4.Drop()
 		defer db4.Close()
 
-		subCtx, subCancel := context.WithTimeout(ctx, time.Second)
+		subCtx, subCancel := context.WithTimeout(ctx, 5*time.Second)
```
Comment (aeddi) [inline tests/replicate_automatically_test.go:183]: Everything else in PR is ok but why did you increase this timeout? You had failures in your tests without it?
Up until now, we hadn't had any problems with that to my knowledge.

Comment (phanhuynhquy) [reply tests/replicate_automatically_test.go:183]: @aeddi : you can see in the commit - for first commit - test is error.
I tested on my Mac then I met same problem if I run 2 tests same time. I debugged then go routines we created after that was not scheduled to run.

Comment (aeddi) [reply tests/replicate_automatically_test.go:183]: Ok 👍

**Review by aeddi — APPROVED** [review]
LGTM 👍

**Pull request merged**